### PR TITLE
Move SlackOAuthController.createSlackTeam to POST handled by SlackCon…

### PR DESCRIPTION
**Pull request by leogrim**
…troller, add introduce slackState parameter

@andrewconner @ryanpbrewster

### CODE REVIEW

File: kifi-backend/modules/shoebox/conf/shoeboxService.routes [hunk 1]
```diff
@@ -17,7 +17,6 @@ GET     /oauth2/slack               @com.keepit.controllers.website.SlackOAuthCo
 
 # These really need to be non-site endpoints
 GET     /site/slack/add                      @com.keepit.controllers.website.SlackOAuthController.addSlackTeam(slackTeamId: Option[SlackTeamId] ?= None)
-GET     /site/organizations/create/slack     @com.keepit.controllers.website.SlackOAuthController.createSlackTeam(slackTeamId: Option[SlackTeamId] ?= None)
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes:20]: Let me do the frontend for this before you merge.

File: kifi-backend/modules/shoebox/app/com/keepit/slack/SlackAuthenticationCommander.scala [hunk 2]
```diff
@@ -87,7 +92,9 @@ class SlackAuthenticationCommanderImpl @Inject() (
       case AddSlackTeam() => slackTeamCommander.addSlackTeam(userId, slackTeamId).map {
         case (slackTeam, isNewOrg) =>
           slackTeam.organizationId match {
-            case None => SlackResponse.RedirectClient(s"/integrations/slack/teams?slackTeamId=${slackTeam.slackTeamId.value}")
+            case None => {
+              SlackResponse.RedirectClient(s"/integrations/slack/teams?slackTeamId=${slackTeam.slackTeamId.value}&slackState=whatever")
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackAuthenticationCommander.scala:96]: `whatever`?

File: kifi-backend/modules/shoebox/conf/shoeboxService.routes (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes:20]: #19638

File: kifi-backend/modules/shoebox/app/com/keepit/slack/SlackAuthenticationCommander.scala (hunk 2 above)
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackAuthenticationCommander.scala:96]: Yeah I'll work on another PR that actually puts something useful in there

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackAuthenticationCommander.scala:96]: Just wanted you to be able to test this

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackAuthenticationCommander.scala:96]: Oh, then can you keep the old routes too? Right now if this is merged, breaks existing site.

**Pull request merged**